app/frontend/testnet/libs/bech32.js

```diff
@@ -0,0 +1,32 @@
+import bech32 from "bech32";
+
+// Note(ppershing): yeah, really, bech32 lib has this stupid api
+export const bech32_encode = (prefix, data) => {
+  const words = bech32.toWords(data);
+  return bech32.encode(prefix, words, 1000);
+};
+
+export const bech32_decode = str => {
+  const tmp = bech32.decode(str, 1000);
+
+  return {
+    prefix: tmp.prefix,
+    data: Buffer(bech32.fromWords(tmp.words))
+  };
+};
+
+export const buf2hex = arrayBuffer => {
+  return Array.from(new Uint8Array(arrayBuffer))
+    .map(n => n.toString(16).padStart(2, "0"))
+    .join("");
+};
+
+export const parse_bech32 = str => {
```

Comment:
this should not be needed after we debug all problems, only `bech32_decode` should be used